test/functional/mempool_wtxid.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python3
+# Copyright (c) 2021 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""
+Test mempool acceptance in case of an already known transaction
+with identical non-witness data different witness.
+"""
+
+from test_framework.script import (
+    CScript,
+    OP_0,
+    OP_TRUE,
+    OP_IF,
+    OP_HASH160,
+    OP_EQUAL,
+    OP_ELSE,
+    OP_ENDIF,
+    hash160,
+)
+from test_framework.messages import (
+    CTransaction,
+    CTxIn,
+    CTxInWitness,
+    CTxOut,
+    COutPoint,
+    sha256,
+    COIN,
+    ToHex,
+)
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+)
+
+class MempoolWtxidTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 1
+        self.setup_clean_chain = True
+        self.extra_args = [["-incrementalrelayfee=0"]]
+
+    def run_test(self):
+        node = self.nodes[0]
+
+        self.log.info('Start with empty mempool, and 200 blocks')
+        self.privkeys = [node.get_deterministic_priv_key().key]
+        self.address = node.get_deterministic_priv_key().address
+        self.coins = []
```

Comment:
No need for these to be member variables:

```suggestion
        privkeys = [node.get_deterministic_priv_key().key]
        address = node.get_deterministic_priv_key().address
        coins = []
```

Reply:
Done - got rid of some of it entirely since not needed for `generate()`